packages/runtime-core/__tests__/rendererSuspense.spec.ts

```diff
@@ -711,5 +713,142 @@ describe('renderer: suspense', () => {
     )
   })
 
-  test.todo('portal inside suspense')
+  test('portal inside suspense', async () => {
+    // TOODO: Remove this. Implementation detail of querySelector in tests
+    resetOps()
+
+    const ChildA = createAsyncComponent({
+      setup() {
+        return () => h(Portal, { target: '#target' }, 'portal content')
+      }
+    })
+
+    const Comp = {
+      setup() {
+        return () => [
+          h(Suspense, null, {
+            default: [h(ChildA)],
+            fallback: h('div', 'root fallback')
+          }),
+          h('div', { id: 'target' })
+        ]
+      }
+    }
+
+    const root = nodeOps.createElement('div')
+    render(h(Comp), root)
+    expect(serializeInner(root)).toBe(
+      `<!----><div>root fallback</div><div id="target"></div><!---->`
+    )
+
+    await Promise.all(deps)
+    await nextTick()
+    expect(serializeInner(root)).toBe(
+      `<!----><!----><!--portal content--><!----><div id=\"target\">portal content</div><!---->`
+    )
+  })
+
+  test('async inside portal inside suspense', async () => {
+    // TOODO: Remove this. Implementation detail of querySelector in tests
+    resetOps()
+
+    const AsyncChildInPortal = createAsyncComponent({
+      setup() {
+        return () => h('div', 'portal content')
+      }
+    })
+
+    const AsyncParentWithPortal = createAsyncComponent({
+      setup() {
+        return () => h(Portal, { target: '#target' }, [h(AsyncChildInPortal)])
+      }
+    })
+
+    const Comp = {
+      setup() {
+        return () => [
+          h(Suspense, null, {
+            default: [h(AsyncParentWithPortal)],
+            fallback: h('div', 'root fallback')
+          }),
+          h('div', { id: 'target' })
+        ]
+      }
+    }
+
+    const root = nodeOps.createElement('div')
+    render(h(Comp), root)
+    expect(serializeInner(root)).toBe(
+      `<!----><div>root fallback</div><div id="target"></div><!---->`
+    )
+
+    await deps[0]
+    await nextTick()
+    expect(serializeInner(root)).toBe(
+      `<!----><div>root fallback</div><div id=\"target\"><!----></div><!---->`
+    )
+
+    await deps[1]
+    await nextTick()
+
+    // TODO: the text node of the comment is an array of vnodes. I don't think this is correct.
+    expect(serializeInner(root)).toBe(
+      `<!----><!----><!--[object Object]--><!----><div id=\"target\"><!----><div>portal content</div></div><!---->`
```

Comment:
`[object Object]` appearing in a comment node feels like either a leaky abstraction or something not being set quite right.